File: backend/app/models/workspace.py
from pydantic import BaseModel


class HealthResponse(BaseModel):
    status: str
    service: str
    version: str


class WorkspaceValidationRequest(BaseModel):
    workspacePath: str


class WorkspaceValidationResponse(BaseModel):
    valid: bool
    exists: bool
    isDirectory: bool
    readable: bool
    normalizedPath: str
    message: str

File: football-collection-builder/frontend/src/types/workspace.ts
export type WorkspaceStatus = 'unconfigured' | 'configured'

export type Workspace = {
  path: string
  status: WorkspaceStatus
}

export type WorkspaceStorageEntry = {
  workspacePath: string
  savedAt: string
  version: number
  status: WorkspaceStatus
}

export type WorkspaceInfo = {
  objective: string
  description: string
  exampleStructure: string[]
}

export type WorkspaceValidationResponse = {
  valid: boolean
  exists: boolean
  isDirectory: boolean
  readable: boolean
  normalizedPath: string
  message: string
}
